### PR TITLE
feat: improved grammar; less (no?) ambiguities;

### PR DESCRIPTION
**Pull request by alacambre-yseop**
Should parse the majority of YML files completely, except text granules (content is skipped).
Assumes that devs use a semi-colon at the end of most of the instructions.
→ The extension now displays warnings for parsing problems.

### CODE REVIEW

File: server/src/YmlToBdl.g4 [hunk 1]
```diff
@@ -218,6 +254,11 @@ fieldValue : field
 
 //Functions
 
+function: (METHOD|FUNCTION|TEXT_METHOD|TEXT_FUNCTION) ymlId ( argsBlock  |  OPEN_PAR argumentList CLOSE_PAR ) localBlock? staticBlock? memberOption=field* SEMICOLON;
```
Comment (mminot-yseop) [inline server/src/YmlToBdl.g4]:
```suggestion
function: (METHOD|FUNCTION|TEXT_METHOD|TEXT_FUNCTION) ymlId (argsBlock|OPEN_PAR argumentList CLOSE_PAR) localBlock? staticBlock? memberOption=field* SEMICOLON;
```

**OR**

```suggestion
function: (METHOD | FUNCTION | TEXT_METHOD | TEXT_FUNCTION) ymlId (argsBlock | OPEN_PAR argumentList CLOSE_PAR) localBlock? staticBlock? memberOption=field* SEMICOLON;
```

?

File: server/src/YmlToBdl.g4 [hunk 2]
```diff
@@ -113,49 +127,57 @@ WS : [ \r\t\n]+ -> channel(HIDDEN);
...
 
 
 //Vérifier adéquation avec les terminaux définis par Olivier
+YMLID: ID (COLON? COLON ID)*;
 ID : LETTER ALPHANUM* ;
 
 fragment MULTILINE_COMMENT_START: '/*';
 fragment MULTILINE_COMMENT_END: '*/';
 
+PREPROCESSING: '@' ~[\n\r]* -> channel(HIDDEN);
```
[the remark this answers is not in the document]
Comment (alacambre-yseop) [reply server/src/YmlToBdl.g4]: I don't know their structure at all, so I prefer to ignore them for now.

File: server/src/YmlToBdl.g4 [hunk 3]
```diff
@@ -113,49 +127,57 @@ WS : [ \r\t\n]+ -> channel(HIDDEN);
...
 expressionMarker : DOT
                  | MULTIVALUED_EXPRESSION;
-preprocessingElement : WS? PREPROCESSING WS?;
-
-freeText : .+? (~EOF)*? ;
-
 
 // Colons aren't interpreted by YE in a particular way.
-ymlId : ID (COLON? COLON ID)*;
+
```
[the remark this answers is not in the document]
Comment (alacambre-yseop) [reply server/src/YmlToBdl.g4]: I changed the comments here and there.

**Comment by alacambre-yseop** [comment]
@mminot-yseop I think this version of the grammar seems stable enough to do new devs upon it and is, therefore, ready for review.
I also added some documentation and updated the `CHANGELOG` file.

File: client/README.md [hunk 4]
```diff
@@ -6,56 +6,58 @@ This extension provides support for Yseop Markup Language (YML).
...
 ### Syntax highlighting
 
-* For common YML code
-* For text granules and inside them
+- For common YML code
+- For text granules and inside them
 
+### Parsing error warnings
+
+![Example image for parsing warning](https://raw.githubusercontent.com/yseop/vscode-yseopml/master/client/images/parsingProblem.png "Parsing warnings in the “Problem”'s and editor's view.")
```
Comment (mminot-yseop) [inline client/README.md]:
```suggestion
![Example image for parsing warning](https://raw.githubusercontent.com/yseop/vscode-yseopml/master/client/images/parsingProblem.png "Parsing warnings in the “Problems” and editor views.")
```

File: client/README.md [hunk 5]
```diff
@@ -6,56 +6,58 @@ This extension provides support for Yseop Markup Language (YML).
...
+- `\senderAction`: Verb using a first person subject
+  **_or_** Verb using a first person subject, with adjective
+- `\switch`: Handle values differently
+- `synonyms`: List of alternatives
+- `TextFunction`: Text function
+  **_or_** Text function with custom arguments
+- `textu`: Textualization attribute
+- `\thirdAction`: Verb using a third person subject
+- `\value`: Display a value
+- `\verb`: Verb
```
Comment (mminot-yseop) [inline client/README.md]: Why change the bullets here and not in the script whose purpose is to generate the list?

Comment (alacambre-yseop) [reply client/README.md]: Automatic formatting, sorry; this was an unexpected change.

**Comment by mminot-yseop** [comment]
Haha, honestly, I was curious to see how you would handle that hyphen affair.

```
$ printf '- `%s`: %s\n' foo bar
bash: printf: - : invalid option
printf: usage: printf [-v var] format [arguments]
```

Hehe. Remember that `--` trick to say “no options from here onwards, only positional arguments”? Now's the time to use it. :laughing:

**Comment by alacambre-yseop** [comment]
@mminot-yseop It didn't failed on my computer, that's why I pushed it.

**Comment by alacambre-yseop** [comment]
Multiple misclicks, sorry.

**Comment by mminot-yseop** [comment]
> @mminot-yseop It didn't failed on my computer, that's why I pushed it.

That's rather strange.
Hum… It seems `xargs` somehow makes that work. I guess it's safer to add the `--` anyway to be more implementation-independent or something.

**Comment by alacambre-yseop** [comment]
@mminot-yseop Done.

**Comment by alacambre-yseop** [comment]
@mminot-yseop It's OK for me in its current state.

**Comment by mminot-yseop** [comment]
I'll try opening random files with it, then, I guess.

**Comment by mminot-yseop** [comment]
Now that I think about it, it's kinda weird to make references to Enhancement Fridays in branch names in a public project!

**Comment by mminot-yseop** [comment]
Hum, so since you ignore `@foo` lines, there is no check that, for example, `@if` has a corresponding `@endif`. (My VSC opened on an old `project.kao` file that was full of these.)

**Comment by mminot-yseop** [comment]
```
```

**Comment by alacambre-yseop** [comment]
@mminot-yseop `project.kao`-like files and preprocessing instructions aren't handled yet but will be in the future.
I added an option in case the user wants to disable the warnings.

**Pull request merged**